spec/facility_spec.rb

```diff
@@ -23,4 +34,105 @@
       expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
     end
   end
+
+# VEHICLE REGISTRATION
+  describe 'regular vehicle registration' do
+    it 'can register a regular vehicle' do
+      @facility_1.add_service("Register Vehicles")
+      expect(@facility_1.collected_fees).to eq(0)
+      @facility_1.register_vehicle(@cruz)
+
+      expect(@cruz.registration_date).to eq(Date.today)
+      expect(@cruz.plate_type).to eq(:regular)
+      expect(@facility_1.registered_vehicles).to eq([@cruz])
+      expect(@facility_1.collected_fees).to eq(100)
+    end
+  end
+
+  describe 'vintage vehicle registration' do
+    it 'can register a vintage vehicle' do
+      @facility_1.add_service("Register Vehicles")
+      @facility_1.register_vehicle(@cruz)
+      expect(@facility_1.collected_fees).to eq(100)
+      @facility_1.register_vehicle(@camaro)
+
+      expect(@camaro.registration_date).to eq(Date.today)
+      expect(@camaro.plate_type).to eq(:antique)
+      expect(@facility_1.registered_vehicles).to eq([@cruz, @camaro])
+      expect(@facility_1.collected_fees).to eq(125)
+    end
+  end
+
+  describe 'electric vehicle registration' do
+    it 'can register an electric vehicle' do
+      @facility_1.add_service("Register Vehicles")
+      @facility_1.register_vehicle(@cruz)
+      expect(@facility_1.collected_fees).to eq(100)
+      @facility_1.register_vehicle(@camaro)
+      expect(@facility_1.registered_vehicles).to eq([@cruz, @camaro])
+      @collected_fees = 125
+      @facility_1.register_vehicle(@bolt)
+
+      expect(@bolt.registration_date).to eq(Date.today)
+      expect(@bolt.plate_type).to eq(:ev)
+      expect(@facility_1.registered_vehicles).to eq([@cruz, @camaro, @bolt])
+      expect(@facility_1.collected_fees).to eq(325)
+    end
+  end
+  
+# GETTING A DRIVER'S LICENSE
+  describe 'getting a drivers license' do
+    it 'takes the written test' do
+      expect(@registrant_1.license_data).to eq({:written => false, :license => false, :renewed => false})
+      expect(@registrant_1.permit?).to be true 
+      @facility_1.add_service("Written Test")
+      @facility_1.administer_written_test?(@registrant_1)
+      expect(@facility_1.administer_written_test?(@registrant_1)).to be true
+      expect(@registrant_1.license_data).to eq({:written => true, :license => false, :renewed => false})
+      
+      expect(@registrant_2.license_data).to eq({:written => false, :license => false, :renewed => false})
+      expect(@registrant_2.age).to eq(16)
+      expect(@registrant_2.permit?).to be false
+      expect(@facility_1.administer_written_test?(@registrant_2)).to be false
+      @registrant_2.earn_permit
+      expect(@facility_1.administer_written_test?(@registrant_2)).to be true
+      expect(@registrant_2.license_data).to eq({:written => true, :license => false, :renewed => false})
```

Comment:
Good job testing for the different conditions that need to be met in order to get the written test. Feel free to add more it blocks for each condition that is being tested, that will help with reading tests.